base_name_search_improved/models/ir_model.py

```diff
@@ -21,6 +23,38 @@ def _get_rec_names(self):
     return rec_name + other_names
 
 
+@tools.ormcache(skiparg=0)
+def _get_rec_exact_names(self):
+    "List of fields to exact search into"
+    model = self.env['ir.model'].search(
+        [('model', '=', str(self._model))])
+    other_names = model.name_search_exact_ids.mapped('name')
+    return other_names
+
+
+@tools.ormcache(skiparg=0)
+def _get_add_smart_search(self):
+    "Add Smart Search on search views"
+    return self.env['ir.model'].search(
+        [('model', '=', str(self._model))]).add_smart_search
+
+
+@tools.ormcache(skiparg=0)
+def _get_name_search_domain(self):
+    "Add Smart Search on search views"
+    name_search_domain = self.env['ir.model'].search(
+        [('model', '=', str(self._model))]).name_search_domain
+    if name_search_domain:
+        return literal_eval(name_search_domain)
+    return []
+
+
+@tools.ormcache(skiparg=0)
+def _get_separator(self):
+    return self.env['ir.config_parameter'].get_param(
```

Comment:
The method `get_param` of `ir.config_parameter` model for v9.0 has lru cache from original modules (is not the case for 8.0)
[ir_config_parameter.py#L70](https://github.com/odoo/odoo/blob/a71cdd02ec6f0a738b48f0bf792de81e17ce240c/openerp/addons/base/ir/ir_config_parameter.py#L70)
Then maybe we don't need a ormcache here.